### DISCUSSION TITLE
Fold Low's Seg2Cat into Core

### DISCUSSION THREAD

**Discussion by liamcrean** [ExpressionEngine/ExpressionEngine#2901] [Feature Requests]
This is brilliant functionality - saves a lot of time pulling category data from urls - can't fault it and one of the first addons added in every build.

https://gotolow.com/addons/low-seg2cat

Would love to see this as part of the core CMS.

**Comment by iammattmartin** [comment]
I agree. Would be a bit of a nightmare if this was ever not updated to support later versions of EE.

**Comment by robsonsobral** [comment]
Hi!

Some of that functionality can be achieved by [this PR](https://github.com/ExpressionEngine/ExpressionEngine/pull/1179)

**Comment by litzinger** [comment]
Having the category variables as early parsed global variables offers advantages over using another tag pair. The global vars can more easily be used in conditionals and as parameters in channel:entries tags. I too personally wouldn't mind seeing a lot of what Low Seg2Cat has as part of the core, and maybe even strategically add some hooks so other 3rd party add-ons can modify the variables (I'm thinking of Publisher and Transcribe, where translated categories appear in the URLs).

Having these variables natively would go a long way toward making categories easier to work with in EE.

Since this functionality does add additional DB queries to a page load it could be an opt-in if it was part of the core and be enabled by a site config value. Could also consider caching the results of the tags so they don't require fresh DB queries each page load, and clearing the cached items every time a category is saved.

**Comment by robsonsobral** [comment]
What do you think of something like a modifier, @litzinger ?

```
{segment_n:category_id}
{segment_n:category_name}
{segment_n:category_description}
{segment_n:category_image}
{segment_n:category_parent_id}
{segment_n:category_group_id}
{last_segment:category_id}
{last_segment:category_name}
{last_segment:category_description}
{last_segment:category_image}
{last_segment:category_parent_id}
{last_segment:category_group_id}
```

So, the DB queries will be done on demand. Or... To reduce queries:

```
{segment_n:category:id}
{segment_n:category:name}
{segment_n:category:description}
{segment_n:category:image}
{segment_n:category:parent_id}
{segment_n:category:group_id}
```

All category data could be get at once and cached.

**Reply by liamcrean** [reply]
This would be ideal

**Comment by litzinger** [comment]
I personally like the 2nd one better with modifiers. There are a lot of other helpful variables that could be added to the core too. I'm not advocating for a clone of URL Helper, but variables like `{segment_category_ids}`, `{segment_category_ids_any}` make it really easy to fetch entries if there are more than 1 category url_title in the url. The `:snake` modifiers are also very helpful. It turns `category/this-is-my-category-url-title` into `{this_is_my_category_url_title}` - for those who prefer to use dashes in URLs, but underscores/snake-case in variable names.

https://github.com/litzinger/URL-Helper

**Comment by robsonsobral** [comment]
I can't see the point of the case changing helpers. Why would you have URLs different from category short names?

---

So, what do you think? Let's invest some time on these modifiers? Is there some "adult" on the room to approve that?

**Comment by litzinger** [comment]
The snake-case implementation was very specific to a project I was working on, so maybe it isn't widely applicable.

**Comment by robsonsobral** [comment]
What about a 1st party [single tag](https://docs.expressionengine.com/latest/development/plugins.html#two-kinds-of-tags) addon, to avoid additional queries? Something like `exp:category`:

```
{exp:channel:entries category="not {exp:category title='{segment_3}'}" parse="inward"}
...
{/exp:cahennel:entries}
```
